src/zx/FunctionalityConstruction.cpp

```diff
@@ -223,6 +226,65 @@ FunctionalityConstruction::parseOp(ZXDiagram& diag, op_it it, op_it end,
       addZSpider(diag, target2, qubits, PiExpression(), EdgeType::Hadamard);
       break;
     }
+    case qc::OpType::RZZ: {
+      const auto ctrl = target;
+      const auto target2 = static_cast<zx::Qubit>(p.at(op->getTargets()[1]));
+      const auto param = parseParam(op.get(), 0);
+
+      addZSpider(diag, ctrl, qubits);
+      addZSpider(diag, target2, qubits);
+
+      auto mid_x = diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)),
+                                  zx::VertexType::X);
+      auto mid_z = diag.addVertex(-1, -1, param, zx::VertexType::Z);
```

Comment:
Couldn't both of these be const? Or are they modified by the `addEdge` method?

Reply:
Yes.